### ISSUE TITLE
adding notifications stops the script before entering 2FA

### ISSUE THREAD

**Issue by brunoamaral** [icloud-photos-downloader/icloud_photos_downloader#685]
## Overview
Adding notifications with smtp stops the script after entering a password and before we have time to enter de 2FA.

## Steps to Reproduce
```
sudo docker run -it --rm --name icloudpd -v /mnt/Photos/originals:/data -e TZ=Europe/Lisbon icloudpd/icloudpd:latest icloudpd --until-found 200 --directory /data --username USER@ICLOUD.com --watch-with-interval 1800 --smtp-host smtp.eu.mailgun.org --notification-email XXXXX@XXXXXX --smtp-username XXXXX@XXXXX --smtp-password PASSWORD
```

## Expected Behavior
Wait for the user to input the 2FA before proceeding

## Actual Behavior
```
2023-09-11 17:50:50 DEBUG    Authenticating...
iCloud Password:
2023-09-11 17:50:55 INFO     Sending 'two-step expired' notification via email...
2023-09-11 17:50:55 INFO     Sending 'two-step expired' notification via email...
script exits
```

## Context
[NOTE]: # ( Give us any additional information you may have. )

**Comment by paulpfeister** [comment]
Experiencing this issue as well. Pretty odd. Also tested with mailgun.

**Comment by AndreyNikiforov** [comment]
The behavior to exit if notification is requested, has been in the code base for some time (==not a recent change).

What is the use case for email notification? If 2fa is required, then Apple is showing alert on a relevant device without extra action from `icloudpd`, so user is notified. I assume that 2fa request has some expiration, so if user has not responded to alert, authentication needs to start all over. From that perspective, makes sense to exit when sending notification with expected long response time (email response is presumably slower than alert on a device). I do not know if Apple behavior for 2fa alerts were different when email notification for `icloudpd` was implemented.

**Comment by npoles** [comment]
FWIW, Apple notifies when the 2FA has expired, but it does not explicitly state which service has expired. Thus, it's actually a little scary to get a random MFA prompt on all of your Apple devices saying some service has requested access to your account. I would not consider the Apple MFA notification a sufficient warning to a user; in fact, the MFA expiry notice should come in advance of the expiration so that users have a chance to get ahead of the expiration and prevent a service interruption.

**Comment by AndreyNikiforov** [comment]
> FWIW, Apple notifies when the 2FA has expired, but it does not explicitly state which service has expired. Thus, it's actually a little scary to get a random MFA prompt on all of your Apple devices saying some service has requested access to your account (especially when the re-auth is potentially not initiated in the moment by a user). I would not consider the Apple MFA notification a sufficient warning to a user; in fact, the MFA expiry notice should come _**in advance**_ of the expiration so that users have a chance to get ahead of the expiration and prevent a service interruption.

I agree that having MFA notifications in advance is better. I don't think it is possible though. Today, 1) Apple sends popup to all devices 2) icloudpd gets response that MFA required 3) icloudpd send notification over email. Besides the order of events, email will always be slower and setting iOS notification on emails (to give context on MFA alert) is not a common practice AFAIK. 

Web UI with local native notification may be helpful (==local iOS notification will popup right at the time of MFA alert and give context), but that path needs research/verification first (and will probably require extra setup).
